File: src/rust/min_max_recursivo.rs
fn main() {
    let vetor = vec![54,42,11,33,24,99,77,80];
    min_max(vetor, i32::MAX, i32::MIN, 0);
}

fn min_max(vetor: Vec<i32>, mut min: i32, mut max: i32, indice: usize) {
    if vetor[indice] < min {
        min = vetor[indice];
    }
    if vetor[indice] > max {
        max = vetor[indice];
    }
    if indice < vetor.len() - 1 {
        min_max(vetor, min, max, indice + 1);
    } else {
        println!{"Data: {:?}", vetor};
        println!{"Min.: {:?}", min};
        println!{"Max.: {:?}", max};
    }
}

File: src/scala/MinMaxRecursivo.scala

def minMax(data: List[Int], minimum: Int, maximum: Int, index: Int): (Int, Int) = {
  var min: Int = minimum
  var max: Int = maximum

  if (data(index) < min) {
    min = data(index)
  }
  if (data(index) > max) {
    max = data(index)
  }

  if (index < data.length - 1) {
    minMax(data, min, max, index + 1)
  } else {
    (min, max)
  }
}

def minMax(data: List[Int]): (Int, Int) = minMax(data, data(0), data(0), 0)

object Main extends App {
  val data: List[Int] = List(4, 6, 2, 9, 3, 8, 1, 7, 5)
  val (min, max) = minMax(data)
  println(s"$data\nMin: $min\nMax: $max")
}
